### PR TITLE
SNOW-879809: Bumped up PythonConnector MINOR version from 3.0.4 to 3.1.0

### PR DESCRIPTION
**Pull request by sfc-gh-anavalos**
@noreview - This is an automated process. No review is required

Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: DESCRIPTION.md [hunk 1]
```diff
@@ -9,6 +9,48 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
...
+   - 6af61ea1 Update cryptography requirement from <41.0.0,>=3.1.0 to >=3.1.0,<42.0.0 (#1590)
+   - a40cf2b9 SNOW-834415 Dont auto create config folder (#1599)
+   - Fix test by adding exception handling for grant query (#1597)
+   - d78a8da9 SNOW-756493 Add param to keep server session alive (#1595)
+   - 48aa932a SNOW-799408: Close temp file handle before removing temp cache file (#1582)
+   - f2d398a6 SNOW-826794 fix mac tests (#1583)
+   - 678f09a7 SNOW-802807 ConfigManager and config file support (#1549)
+   - Revert changes related to raw json (#1577)
+   -
+
```
Comment (sfc-gh-aling) [inline DESCRIPTION.md:53]: I don't think we need this part, all the changes are already in the description below

File: DESCRIPTION.md [hunk 2]
```diff
@@ -9,6 +9,48 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
 # Release Notes
 
 - v3.0.5(TBD)
+
```
Comment (sfc-gh-aling) [inline DESCRIPTION.md:12]:
```suggestion
```

Comment (sfc-gh-aling) [reply DESCRIPTION.md:12]: deduplicate the version in changelog

File: DESCRIPTION.md [hunk 3]
```diff
@@ -8,7 +8,7 @@ Source code is also available at: https://github.com/snowflakedb/snowflake-conne
 
 # Release Notes
 
-- v3.0.5(TBD)
+- v3.0.5(July 31,2023)
```
Comment (sfc-gh-aling) [inline DESCRIPTION.md:11]: @sfc-gh-mkeller , I remember you want the version to be v3.1.0?

Comment (sfc-gh-mkeller) [inline DESCRIPTION.md:11]:
```suggestion
- v3.1.0(July 31,2023)
```

File: src/snowflake/connector/version.py [hunk 4]
```diff
@@ -1,3 +1,3 @@
 # Update this for the versions
 # Don't change the forth version number from None
-VERSION = (3, 0, 4, None)
+VERSION = (3, 0, 5, None)
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/version.py:3]:
```suggestion
VERSION = (3, 1, 0, None)
```

**Review by sfc-gh-mkeller — CHANGES_REQUESTED** [review]

**Review by sfc-gh-mkeller — APPROVED** [review]
:ship: :ship:

**Review by sfc-gh-aling — APPROVED** [review]

**Pull request merged**